app/components/Reactions/ReactionPicker.js

```diff
@@ -21,6 +22,120 @@ type Props = {
+  currentEmojis = currentEmojis.filter(emoji => {
+    if (`:${searchString}:` === emoji.shortCode.toLowerCase()) {
+      matchingEmojis.push(emoji);
+      return false;
+    }
+    {
+      return true;
+    }
+  });
+
+  /*
+   * STEP 2
+   * Then look if shortCode start with, ends with or contains the searchString in the middle.
+   *
+   * Examples (searchString = "car"):
+   *   - ":car_test:" (match = ":<car_>test:")
+   *   - ":test_car:" (match = ":test<_car>:")
+   *   - ":good_car_test:" (match = ":good<_car_>test:")
+   */
+  currentEmojis = currentEmojis.filter(emoji => {
+    const startsWithExact = emoji.shortCode
+      .toLowerCase()
+      .startsWith(`:${searchString}_`);
+    const endsWithExact = emoji.shortCode
+      .toLowerCase()
+      .endsWith(`_${searchString}:`);
+    const contains = emoji.shortCode
+      .toLowerCase()
+      .includes(`_${searchString}_`);
+    if (startsWithExact || endsWithExact || contains) {
+      matchingEmojis.push(emoji);
+      return false;
+    } else {
+      return true;
+    }
+  });
+
+  /*
+   * STEP 3
+   * Then look if shortCode start with, ends with or contains the searchString in the middle.
+   *
+   * Examples (searchString = "car"):
+   *   - ":car2:" (match = ":<car>2:")
+   *   - ":carrot:" (match = ":<car>rot")
+   *   - ":2car:" (match = ":2<car>:")
+   */
+  currentEmojis = currentEmojis.filter(emoji => {
+    const startsWith = emoji.shortCode
+      .toLowerCase()
+      .startsWith(`:${searchString}`);
+    const endsWith = emoji.shortCode.toLowerCase().endsWith(`${searchString}:`);
+    if (startsWith || endsWith) {
+      matchingEmojis.push(emoji);
+      return false;
+    } else {
+      return true;
+    }
+  });
+
+  /*
+   * STEP 3
+   * Look if the keywords for the emoji match exactly
+   *
+   * Examples (searchString = "vehicle"):
+   *   - ":taxi:" (match = "vehicle", keywords = ["uber", "vehicle", "cars", "transportation"])
+   *   - ":tractor:" (match = "vehicle", keywords = ["vehicle", "car", "farming", "argiculture"])
+   *   - ":oncoming_automobile:" (match = "vehicle", keywords = ["car", "vehicle", "transportation"])
+   */
+  currentEmojis = currentEmojis.filter(emoji => {
+    if (
+      emoji.keywords.find(keyword => keyword.toLowerCase() == searchString) !==
+      undefined
+    ) {
+      matchingEmojis.push(emoji);
+    } else {
+      return true;
+    }
+  });
+
+  /*
+   * STEP 4
+   * Perform a fuzzy search at the end for the remaining emojis.
+   *
+   * Examples (searchString = "car"):
+   *   - ":woman_cart_wheeling:" (match = ":woman_<car>t_wheeling:")
+   *   - ":costa_rica:" (match = ":<c>ost<a>_<r>ica")
+   *   - ":christmas_tree:" (match = ":<c>hristm<a>s_t<r>ee")
+   */
+  var results = fuzzy.filter(searchString, currentEmojis, {
+    pre: '<',
+    post: '>',
+    extract: emoji => {
+      return emoji.shortCode;
+    }
+  });
+
+  currentEmojis = results.forEach(result => {
+    matchingEmojis.push(result.original);
+  });
```

Comment:
It seems a bit unnecessary to modify `currentEmojis`, can't you instead do something like:
```js
return [...results.map(result => result.original), ...matchingEmojis]
```
I'm not sure what the performance difference is though.